lib/common/xml.c

```diff
@@ -1463,9 +1484,11 @@ pcmk__xml_escape(const char *text, bool escape_quote)
     pcmk__str_update(&copy, text);
 
     for (size_t index = 0; index < length; index++) {
-        if (((copy[index] & 0x80) != 0) && ((copy[index + 1] & 0x80) != 0)) {
-            /* @TODO Is this a valid test for multi-byte Unicode characters?
-             * Probably better to escape them anyway.
+        if (is_unicode(&copy[index])) {
+            /* Skip two bytes (one here, one in loop counter).
+             *
+             * @TODO Better to escape these? But readability concerns for
+             * languages with non-ASCII alphabets.
```

Comment:
There shouldn't be any reason to escape them, as long as we're always using UTF-8 encoding, which I believe we are, though we probably could do more to advertise it correctly, in particular using the XML prolog (`<?xml version = "1.0" encoding = "UTF-8" standalone = "no" ?>`)

Reply:
> no reason

Nice. We should be *able* to escape them, but escaping them (albeit incorrectly) was why that upstream contributor added the "skip" block in the first place. It was making Chinese text unreadable.

If it's valid to use UTF-8 characters, I'd prefer to do so.

> prolog

Added to description of T767.